include/bluetooth/gatt.h

```diff
@@ -1340,13 +1356,27 @@ struct bt_gatt_discover_params {
  *  For each attribute found the callback is called which can then decide
  *  whether to continue discovering or stop.
  *
- *  @note This procedure is asynchronous therefore the parameters need to
- *        remains valid while it is active.
+ *  The Response comes in callback @p params->func. The callback is run from
+ *  the BT RX thread. @p params must remain valid until start of callback where
+ *  iter `attr` is `NULL` or callback will return `BT_GATT_ITER_STOP`.
+ *
+ *  This function will block while the ATT request queue is full, except when
+ *  called from the BT RX thread, as this would cause a deadlock.
  *
  *  @param conn Connection object.
  *  @param params Discover parameters.
  *
  *  @return 0 in case of success or negative value in case of error.
+ *
+ *  @retval 0
+ *      Successfully queued request.
+ *      Will call @p params->func on resolution.
+ *
+ *  @retval -ENOMEM
+ *      ATT request queue is full and blocking would cause deadlock.
+ *      Allow a pending request to resolve before retrying, or call this
+ *      function outside the BT RX thread to get blocking behavior.
+ *      Queue size is controlled by @kconfig{CONFIG_BT_L2CAP_TX_BUF_COUNT}.
```

Comment:
this documentation is now inconsistent, looks like first `@return` is now redundant?

better use this format, for consistency:
```c
 *  @retval -ENOMEM ATT request queue is full and blocking would cause deadlock.
 *  Allow a pending request to resolve before retrying, or call this function outside the
 *  BT RX thread to get blocking behavior. Queue size is controlled by 
 *  @kconfig{CONFIG_BT_L2CAP_TX_BUF_COUNT}.
```